pkg/typesystem/typesystem.go

```diff
@@ -778,6 +809,85 @@ func (t *TypeSystem) validateRelationTypeRestrictions() error {
 	return nil
 }
 
+// ensureNoCyclesInTupleToUsersetDefinitions throws an error on the following models because `viewer` is a cycle.
+// type folder
+//
+//	 relations
```

Comment:
Mind fixing indentation in this godoc?

Should only need 2 indentations per level.

Reply:
Goland is messing this up before I commit :(

Reply:
I used VS Code to fix it but then the linter complained so i had to revert

Reply:
I can fix it and push.